File: code/python/isIsomorphic.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        if s == t:
            return True
        dict1 = {}
        dict2 = {}
        for i in range(len(s)):
            if s[i] not in dict1:
                dict1[s[i]] = [i]
            else:
                dict1[s[i]].append(i)
            if t[i] not in dict2:
                dict2[t[i]] = [i]
            else:
                dict2[t[i]].append(i)

        for i in range(len(s)):
            if dict1[s[i]] != dict2[t[i]]:
                return False
        
        return True
        

File: code/java/isIsomorphic.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length())
            return false;
        if(s.equals(t))
            return true;
        Map<Character,String> ms = new HashMap<Character,String>();
        Map<Character,String> mt = new HashMap<Character,String>();
        String temp;
        for(int i = 0; i < s.length(); i++){
            if(!ms.containsKey(s.charAt(i)))
                ms.put(s.charAt(i),i+"");
            else{
                temp = ms.get(s.charAt(i));
                ms.replace(s.charAt(i),temp+(i + ""));
            }
            if(!mt.containsKey(t.charAt(i)))
                mt.put(t.charAt(i),i+"");
            else{
                temp = mt.get(t.charAt(i));
                mt.replace(t.charAt(i),temp+(i + ""));
            }
        }
        
        for(int i = 0; i < s.length(); i++)
            if(!ms.get(s.charAt(i)).equals(mt.get(t.charAt(i))))
                return false;
        return true;
    }
}
